fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/util/TrieNode.java

```diff
@@ -77,4 +78,22 @@ public Collection<T> values() {
                 .flatMap(trie -> trie.getValue() == null ? Stream.of() : Stream.of(trie.getValue()))
                 .collect(ImmutableList.toImmutableList());
     }
+
+    @Override
+    public boolean equals(final Object other) {
+        if (this == other) {
+            return true;
+        }
+        if (!(other instanceof TrieNode)) {
```

Comment:
Could this compare `getClass` instead and eliminate the need for the subclass to override?

Reply:
Done.